storage_service/locations/api/resources.py

```diff
@@ -839,6 +840,22 @@ def aip_store_callback_request(self, request, bundle, **kwargs):
         else:
             return http.HttpNoContent()
 
+    @_custom_endpoint(expected_methods=['post'])
+    def reindex_request(self, request, bundle, **kwargs):
+        package = bundle.obj
+        if package.package_type != Package.TRANSFER:
+            return http.HttpBadRequest(json.dumps({"error": True, "message": "This package is not a transfer."}), content_type='application/json')
+        if package.current_location.purpose != Location.BACKLOG:
+            return http.HttpBadRequest(json.dumps({"error": True, "message": "This package is not part of a transfer backlog location."}), content_type='application/json')
```

Comment:
Currently, Transfers only go into Backlog locations, but that may not always be true, so I'd vote to keep the extra check.